### PR TITLE
Reid/through gate

### PR DESCRIPTION
**Pull request by reoliveira**
wrote up a quick subroutine for going through the gate in prep for integration tests

### CODE REVIEW

File: src/decision/CMakeLists.txt [hunk 1]
```diff
@@ -56,7 +56,9 @@ add_executable(decision_node
         subroutines/qualification/include/LocateGate.h
         subroutines/qualification/src/LocateGate.cpp
         subroutines/qualification/include/LineUpWithGate.h
-        subroutines/qualification/src/LineUpWithGate.cpp)
+        subroutines/qualification/src/LineUpWithGate.cpp
+        subroutines/qualification/include/GoThroughGate.h
+        subroutines/qualification/src/GoThroughGate.cpp)
```
Comment (garethellis0) [inline src/decision/CMakeLists.txt]: I missed this back in #36, but noticed a the same thing in #44, so just copying my comment here:
"""
Our convention (and a pretty standard C++ one), is to have the `src` and `include` folders both at top level, ie

`routine/Gate/include/locatingGate.h` -> `include/routine/Gate/locatingGate.h`
`routine/Gate/src/locatingGate.cpp` -> `src/routine/Gate/locatingGate.cpp`
"""
Obviously this isn't a super set-in-stone thing, but it's a fairly normal convention for c++ projects I think

File: src/decision/subroutines/qualification/include/GoThroughGate.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/*
+ * Created By: Reid Oliveira
+ * Created On: May 19, 2018
+ * Description:
```
Comment (garethellis0) [inline src/decision/subroutines/qualification/include/GoThroughGate.h]: We should fill in the "Description" here and in the `.cpp` file

File: src/decision/CMakeLists.txt (hunk 1 above)
Comment (reoliveira) [reply src/decision/CMakeLists.txt]: refactored

File: src/decision/subroutines/qualification/include/GoThroughGate.h (hunk 2 above)
Comment (reoliveira) [reply src/decision/subroutines/qualification/include/GoThroughGate.h]: done

File: src/decision/CMakeLists.txt (hunk 1 above)
Comment (reoliveira) [reply src/decision/CMakeLists.txt]: Yeah, now that I've made this change I can see it's WAY cleaner. Good call.

**Comment by garethellis0** [comment]
LGTM! :+1: :tada:

**Pull request merged**